### PR TITLE
Database thread pool use

### PR DESCRIPTION
**Pull request by PJB3005**
Partially based on a patch by @partmedia

First, we stop re-using DbContexts for SQLite. This apparently fixes performance issues where SQLite stutters like mad. As I understand it EFCore will still do caching internally so this shouldn't hurt performance in any way. This is the partmedia patch.

Secondly, we thread-pool DB operations now.

This means two things:

* SQLite operations don't synchronously block the main thread, and can actually be ran in parallel (for however much SQLite allows this). Concurrency defaults to 3 and can be configured with database.sqlite_concurrency.
* We avoid jumping back to the SynchronizationContext inside the DB layer code, improving performance on Postgres.

### CODE REVIEW

File: Content.Server/Database/ServerDbBase.cs [hunk 1]
```diff
@@ -40,6 +40,8 @@ public abstract class ServerDbBase
                 profiles[profile.Slot] = ConvertProfiles(profile);
             }
 
+            Console.WriteLine(Thread.CurrentThread.Name);
```
Comment (moonheart08) [inline Content.Server/Database/ServerDbBase.cs:43]: debug print!!!!

Comment (PJB3005) [reply Content.Server/Database/ServerDbBase.cs:43]: grug

**Comment by DEATHB4DEFEAT** [comment]
I tried this out with ~15 people and almost everyone had their preferences fail to load, rejoining fixed it for some but most couldn't get it to work.

**Comment by PJB3005** [comment]
Any errors on the server? Also was this on latest master? I think there was an unrelated game bug for a moment.

**Comment by DEATHB4DEFEAT** [comment]
On an older version, I'll check for server errors

**Comment by DEATHB4DEFEAT** [comment]
image

**Comment by PJB3005** [comment]
That honestly almost looks like a bug in EFCore more than anything else. Great.

That said, I did forget to dispose the DbContexts so maybe fixing this will fix that?

**Comment by PJB3005** [comment]
image

Great, EFCore is broken.

**Comment by PJB3005** [comment]
Alright I figured out what the problem is but the fix might take a bit to write because I realize the current database code is a bit "uhhh" with tests.

**Comment by PJB3005** [comment]
@DEATHB4DEFEAT would you mind trying again? Thanks!

**Comment by DEATHB4DEFEAT** [comment]
Waited for the current round to end to update (lasted 5 hours)

Seems to be working fine, will tell you if I encounter any issues later

**Comment by metalgearsloth** [comment]
@PJB3005 You coming back to this.

**Pull request merged**